### ISSUE TITLE
Yolo3 Pascal VOC 2012  loss went to infinite after 10 epochs in fine tuning 

### ISSUE THREAD

**Issue by chulminkw** [david8862/keras-YOLOv3-model-set#168]
Hello, 

I was training Yolo 3 model for Pascal VOC 2012 dataset with the following command. 

python ./trainpy --model_type=yolo3_darknet --anchors_path=configs/yolo3_anchors.txt --annotation_file=../trainval.txt --classes_path=configs/voc_classes.txt --eval_online --save_eval_checkpoint

I used darknet53.conv.74 and convert it to darknet53.h5 by using the following command. 
wget -O weights/darknet53.conv.74.weights https://pjreddie.com/media/files/darknet53.conv.74
python tools/model_converter/convert.py cfg/darknet53.cfg weights/darknet53.conv.74.weights weights/darknet53.h5

But starting from the 11th epoch when unfreezing models layers for the 2nd find-tuning stage, the training and validation loss went to infinite. And training the model was stopped because of TerminateOnNaN Callback. 

I was doing this on a Kaggle kernel which only supports tensorflow 2.4 version. 
I downgraded MNN packages to 1.1.0 as it seems some problems with higher numpy version in kaggle(pip install MNN==1.1.0)

the followings are the model's fit_generator messages. There was online evaluation after 10 epochs but the performance is poor.  How can I correct this?

==============================================================================
327/327 [==============================] - 226s 626ms/step - loss: 997.8475 - location_loss: 1.8531 - confidence_loss: 272.7010 - class_loss: 8.9530 - val_loss: 62.1977 - val_location_loss: 1.8134 - val_confidence_loss: 51.5925 - val_class_loss: 8.5486

Epoch 00001: val_loss improved from inf to 62.19772, saving model to logs/000/ep001-loss283.750-val_loss62.198.h5
Epoch 2/20
327/327 [==============================] - 196s 602ms/step - loss: 35.4629 - location_loss: 1.7971 - confidence_loss: 23.4709 - class_loss: 7.9690 - val_loss: 29.7484 - val_location_loss: 1.7896 - val_confidence_loss: 19.9592 - val_class_loss: 7.7562

Epoch 00002: val_loss improved from 62.19772 to 29.74839, saving model to logs/000/ep002-loss33.480-val_loss29.748.h5
Epoch 3/20
327/327 [==============================] - 192s 588ms/step - loss: 28.9709 - location_loss: 1.7967 - confidence_loss: 18.7416 - class_loss: 7.9461 - val_loss: 30.6881 - val_location_loss: 1.7247 - val_confidence_loss: 20.8604 - val_class_loss: 7.8594

Epoch 00003: val_loss did not improve from 29.74839
Epoch 4/20
327/327 [==============================] - 192s 588ms/step - loss: 26.9876 - location_loss: 1.7569 - confidence_loss: 17.1084 - class_loss: 7.8144 - val_loss: 26.6317 - val_location_loss: 1.6508 - val_confidence_loss: 17.1564 - val_class_loss: 7.5808

Epoch 00004: val_loss improved from 29.74839 to 26.63174, saving model to logs/000/ep004-loss26.923-val_loss26.632.h5
Epoch 5/20
327/327 [==============================] - 191s 586ms/step - loss: 27.1305 - location_loss: 1.7533 - confidence_loss: 16.5187 - class_loss: 7.7965 - val_loss: 26.5190 - val_location_loss: 1.8560 - val_confidence_loss: 16.4597 - val_class_loss: 7.9592

Epoch 00005: val_loss improved from 26.63174 to 26.51900, saving model to logs/000/ep005-loss26.312-val_loss26.519.h5
Epoch 6/20
327/327 [==============================] - 191s 584ms/step - loss: 25.5026 - location_loss: 1.6830 - confidence_loss: 16.0833 - class_loss: 7.6914 - val_loss: 25.0382 - val_location_loss: 1.5536 - val_confidence_loss: 15.8357 - val_class_loss: 7.4046

Epoch 00006: val_loss improved from 26.51900 to 25.03825, saving model to logs/000/ep006-loss25.702-val_loss25.038.h5
Epoch 7/20
327/327 [==============================] - 192s 587ms/step - loss: 25.1650 - location_loss: 1.6279 - confidence_loss: 15.7413 - class_loss: 7.6113 - val_loss: 26.6346 - val_location_loss: 1.6574 - val_confidence_loss: 16.8777 - val_class_loss: 7.8546

Epoch 00007: val_loss did not improve from 25.03825
Epoch 8/20
327/327 [==============================] - 187s 573ms/step - loss: 25.2421 - location_loss: 1.6059 - confidence_loss: 15.6696 - class_loss: 7.6289 - val_loss: 26.3448 - val_location_loss: 1.6481 - val_confidence_loss: 16.7345 - val_class_loss: 7.7169

Epoch 00008: val_loss did not improve from 25.03825
Epoch 9/20
327/327 [==============================] - 188s 576ms/step - loss: 24.3280 - location_loss: 1.6040 - confidence_loss: 15.5303 - class_loss: 7.5376 - val_loss: 24.9575 - val_location_loss: 1.5264 - val_confidence_loss: 15.7065 - val_class_loss: 7.4787

Epoch 00009: val_loss improved from 25.03825 to 24.95751, saving model to logs/000/ep009-loss24.918-val_loss24.958.h5
Epoch 10/20
327/327 [==============================] - 188s 576ms/step - loss: 25.3332 - location_loss: 1.6120 - confidence_loss: 15.7065 - class_loss: 7.5838 - val_loss: 29.4051 - val_location_loss: 1.6129 - val_confidence_loss: 19.7891 - val_class_loss: 7.7565

Epoch 00010: val_loss did not improve from 24.95751
Eval model: 100%|█████████████████████████████| 582/582 [01:06<00:00,  8.71it/s]
/kaggle/working/keras-YOLOv3-model-set/eval.py:630: MatplotlibDeprecationWarning: 
The set_window_title function was deprecated in Matplotlib 3.4 and will be removed two minor releases later. Use manager.set_window_title or GUI-specific methods instead.
  fig.canvas.set_window_title('AP ' + class_name)
/kaggle/working/keras-YOLOv3-model-set/eval.py:777: MatplotlibDeprecationWarning: 
The set_window_title function was deprecated in Matplotlib 3.4 and will be removed two minor releases later. Use manager.set_window_title or GUI-specific methods instead.
  fig.canvas.set_window_title(window_title)

Pascal VOC AP evaluation
dog: AP 0.0292, precision 0.0295, recall 0.1954
person: AP 0.0261, precision 0.0043, recall 0.5965
diningtable: AP 0.0067, precision 0.0287, recall 0.1852
aeroplane: AP 0.0026, precision 0.0667, recall 0.0233
cat: AP 0.0024, precision 0.0248, recall 0.0870
bicycle: AP 0.0000, precision 0.0000, recall 0.0000
bird: AP 0.0000, precision 0.0000, recall 0.0000
boat: AP 0.0000, precision 0.0000, recall 0.0000
bottle: AP 0.0000, precision 0.0000, recall 0.0000
bus: AP 0.0000, precision 0.0000, recall 0.0000
car: AP 0.0000, precision 0.0000, recall 0.0000
chair: AP 0.0000, precision 0.0000, recall 0.0000
cow: AP 0.0000, precision 0.0000, recall 0.0000
horse: AP 0.0000, precision 0.0000, recall 0.0000
motorbike: AP 0.0000, precision 0.0000, recall 0.0000
pottedplant: AP 0.0000, precision 0.0000, recall 0.0000
sheep: AP 0.0000, precision 0.0000, recall 0.0000
sofa: AP 0.0000, precision 0.0000, recall 0.0000
train: AP 0.0000, precision 0.0000, recall 0.0000
tvmonitor: AP 0.0000, precision 0.0000, recall 0.0000
mAP@IoU=0.50 result: 0.335045
mPrec@IoU=0.50 result: 0.769741
mRec@IoU=0.50 result: 5.436672
Epoch 11/20
 58/327 [====>.........................] - ETA: 2:21 - loss: nan - location_loss: nan - confidence_loss: nan - class_loss: nan              Batch 57: Invalid loss, terminating training
327/327 [==============================] - 49s 148ms/step - loss: nan - location_loss: nan - confidence_loss: nan - class_loss: nan - val_loss: nan - val_location_loss: nan - val_confidence_loss: nan - val_class_loss: nan

Epoch 00011: val_loss did not improve from 24.95751
Unfreeze and continue training, to fine-tune.
Train on 5241 samples, val on 582 samples, with batch size 16, input_shape (416, 416).
Epoch 21/250
  1/327 [..............................] - ETA: 2:46:54 - loss: nan - location_loss: nan - confidence_loss: nan - class_loss: nanBatch 0: Invalid loss, terminating training
327/327 [==============================] - 53s 68ms/step - loss: nan - location_loss: nan - confidence_loss: nan - class_loss: nan - val_loss: nan - val_location_loss: nan - val_confidence_loss: nan - val_class_loss: nan

Epoch 00021: val_loss did not improve from 24.95751

**Comment by david8862** [comment]
@chulminkw It seems the darknet backbone pretrained weights are not well trained on Imagenet and couldn't achieve a good training performance with it. You can refer #126

**Comment by chulminkw** [comment]
Thanks, david8862
Is there any chance where loss is divided by 0 or garbage true data is used which leads to infinite loss?
I've checked with no pretrained weight model. But it gives the same infinite loss. 

Best regards.
